text/php/0002-buildpack-yml-to-env-var.md

```diff
@@ -0,0 +1,222 @@
+variable configuration option:
+```yaml
+php:
+ script:
+```
+This structure was effectively taking the part and behavior of [Paketo Procfile
+buildpack](https://github.com/paketo-buildpacks/procfile) by allowing users to
+provide a start command in `buildpack.yml`. However, as part of the [PHP Rewrite RFC](https://github.com/paketo-buildpacks/rfcs/blob/main/text/php/0001-restructure.md)
+the Procfile buildpack will be added to all order groupings. Because of this
+addition I am proposing the removal of this configuration option in the
+environment variables in favor of encouraging users to use a Procfile if they
+need to set a custom start script.
+
+---
+The proposed environment variables for Composer are as follows:
+
+### `BP_COMPOSER_VERSION`
+```shell
+$BP_COMPOSER_VERSION="1.10.x"
+```
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+composer:
+  version: 1.10.x
+```
+### `BP_COMPOSER_INSTALL_OPTIONS`
+```shell
+$BP_COMPOSER_INSTALL_OPTIONS="--no-dev --prefer-install=auto"
+```
+
+Note: This will be parsed using this [shellwords library](https://github.com/mattn/go-shellwords).
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+composer:
+  install_options: ["--no-dev", "--prefer-install=auto"]
+```
+### `BP_COMPOSER_VENDOR_DIR`
+```shell
+$BP_COMPOSER_VENDOR_DIR="vendor"
+```
+
+Note: The value of this environment variable will be used to set
+[`$COMPOSER_VENDOR_DIR`](https://getcomposer.org/doc/03-cli.md#composer-vendor-dir)
+therefore if a user attempts to set this value themselves as well as set the
+value of `$BP_COMPOSER_VENDOR_DIR` the value of `$COMPOSER_VENDOR_DIR` will be
+overwritten resulting in potentially unexpected behavior.
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+composer:
+  vendor_directory: vendor
+```
+
+### `BP_COMPOSER_HOME`
+```shell
+$BP_COMPOSER_HOME="composer"
+```
+
+Note: The value of this environment variable will be used to set
+[`$COMPOSER_HOME`](https://getcomposer.org/doc/03-cli.md#composer-home)
+therefore if a user attempts to set this value themselves as well as set the
+value of `$BP_COMPOSER_HOME` the value of `$COMPOSER_HOME` will be overwritten
+resulting in potentially unexpected behavior.
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+composer:
+  json_path: composer
+```
+
+### `BP_COMPOSER_GLOBAL_INSTALL_OPTIONS`
+```shell
+$BP_COMPOSER_GLOBAL_INSTALL_OPTIONS="--only-name --type"
+```
+
+Note: This will be parsed using this [shellwords library](https://github.com/mattn/go-shellwords).
+
+This will replace the following structure in `buildpack.yml`:
+```yaml
+composer:
+  install_global: ["--only-name", "--type"]
+```
+
+---
+
+### Deprecation Strategy
+In order to facilitate a smooth transition from `buildpack.yml`, the buildpack
+should will support both configuration options with environment variables
+taking priority or `buildpack.yml` until the 1.0 release of the buildpack. The
```

Comment:
A majority of the existent buildpacks in the PHP language family are going to be rewritten following https://github.com/paketo-buildpacks/php/issues/485. Should net-new buildpacks have to implement support for both `buildpack.yml` as well as environment variables to ensure backwards compatibility?

Reply:
I think that is a pretty valid question. My impression was that at the end of the rewrite while we would have different buildpacks the overall functionality should remain the same. If that is not the case and we are already breaking the high level API so to speak then I think that we should deprecate the `buildpack.yml` at the same time as a large API change.

Reply:
The main goal of the rewrite work is meant to increase modularity and maintainability. I think you are right in saying that the functionality should more or less stay the same. However, in https://github.com/paketo-buildpacks/rfcs/blob/main/text/php/0001-restructure.md I don't see any mention of maintaining `buildpack.yml` support..so it isn't clear cut.

I think it probably makes the most sense to include `buildpack.yml` support for backwards compatibility, BUT it pains me since we are largely moving away from `buildpack.yml` support across all of our buildpacks..

Reply:
The plan discussed yesterday:

1. Add deprecation messages for buildpack.yml to the current version of the buildpack. Something like `Use of 'buildpack.yml' has been deprecated and will be removed in the next major branch of the buildpack.`
2. In the new buildpack, we don't support buildpack.yml at all. Including one will generate an error so that it actively fails and users are forced to address it (i.e. don't silently miss this change).
3. In the new buildpack, we will support config through env variables.
4. We'll have a migration doc explaining how to convert settings from buildpack.yml to env variables.

Reply:
Filing issues across PHP for this now

Reply:
https://github.com/paketo-buildpacks/php-web/issues/216
https://github.com/paketo-buildpacks/php-dist/issues/329
https://github.com/paketo-buildpacks/php-composer/issues/265

Reply:
I think this section needs to be rewritten given the plan discussed here.